test/unit/component.js

```diff
@@ -82,142 +85,344 @@ describe('Component class', () => {
-        component.updateConfig({test: 'lol'});
-        assert.calledWith(component.updater.updateConfig, {test: 'lol'});
+      describe('styles', () => {
+        it('returns styles for each component in manifest', () => {
+          component = new Component({
+            id: 1234,
+            options: {
+              product: {
+                styles: {
+                  button: {
+                    color: 'red',
+                  },
+                },
+              },
+            },
+          });
+          component.typeKey = 'product';
+          assert.deepEqual(component.styles, {product: {button: {color: 'red'}}});
+        });
+      });
+
+      describe('classes', () => {
+        it('returns classes for each component in manifest', () => {
+          component.typeKey = 'product';
+          assert.equal(component.classes.product.product, 'shopify-buy__product');
+          assert.equal(component.classes.option.option, 'shopify-buy__option-select');
+        });
+      });
+
+      describe('selectors', () => {
+        it('returns classes formatted as css selectors for each component in manifest', () => {
+          component.typeKey = 'product';
+          assert.equal(component.selectors.product.product, '.shopify-buy__product');
+          assert.equal(component.selectors.option.option, '.shopify-buy__option-select');
+        });
+      });
+
+      describe('googleFonts', () => {
+        it('returns google fonts for each component in manifest', () => {
+          const config = {
+            options: {
+              product: {
+                googleFonts: ['product'],
+              },
+              option: {
+                googleFonts: ['option'],
+              },
+            },
+          };
+          component = new Component(config);
+          component.typeKey = 'product';
+          assert.deepEqual(component.googleFonts, ['product', 'option']);
+        });
+      });
+
+      describe('viewData', () => {
+        it('returns merged object of model, viewData, classes, and text', () => {
+          component.typeKey = 'product';
+          component.model = {model: 'model'};
+          component = Object.defineProperty(component, 'options', {
+            writable: false,
+            value: {
+              viewData: {viewData: 'viewData'},
+              text: 'text',
+            },
+          });
+          component = Object.defineProperty(component, 'classes', {
+            writable: false,
+            value: 'classes',
+          });
+          const expectedObj = {
+            viewData: 'viewData',
+            text: 'text',
+            model: 'model',
+            classes: 'classes',
+          };
+          assert.deepEqual(component.viewData, expectedObj);
+        });
+      });
+
+      describe('morphCallbacks', () => {
+        it('returns an object with the function onBeforeElUpdated', () => {
+          assert.instanceOf(component.morphCallbacks.onBeforeElUpdated, Function);
```

Comment:
Although it's implied, explicitly testing that the getter returns an object with a property that is a function will be better.